docs/content/en/docs/concepts/architecture/components/metrics-operator/_index.md

```diff
@@ -33,22 +46,65 @@ style Metrics-Adapter fill:#d8e6f4,stroke:#fff,stroke-width:px,color:#006bb8
+It is an important component of the metrics operator
+as it allows for the collection and exposure of custom metrics,
+which can be used to gain insight into the behavior and performance
+of applications running on a Kubenetes cluster.
+
+The **Metrics controller** fetches metrics from an SLI provider.
+The controller reconciles a [`KeptnMetric`](../../../../yaml-crd-ref/metric.md)
+resource and updates its status with the metric value
+provided by the selected metric provider.
+Each `KeptnMetric` is identified by `name`
+and is associated with an instance of an observability platform
+that is defined in a
+[KeptnMetricsProvider](../../../../yaml-crd-ref/metricsprovider.md)
+resource.
 
-**Metrics controller** is used to fetch metrics from a SLI provider.
-The controller reconciles a [`KeptnMetric`](../../../../yaml-crd-ref/metric.md) CR and
-updates its status with the metric value provided by the selected SLI provider.
 The steps in which the controller fetches metrics are given below:
 
-* It first fetches the `KeptnMetric` object to reconcile.
-* If the object is not found, it returns and lets Kubernetes handle deleting all associated resources.
-* If the object is found, the code checks that if the metric has been updated within the configured
-interval which is defined in the `Spec.FetchIntervalSeconds`.
-If not, then it skips reconciling and requeues the request for later.
-* If the metric should be reconciled, it fetches the provider defined in the `Spec.Provider.Name` field.
-* If the provider is not found, it returns and requeues the request for later.
-* If the provider is found, it loads the provider and evaluates the query defined in the `Spec.Query` field.
-* If the evaluation is succesful, it stores the fetched value in the status of the `KeptnMetric` object.
+1. Fetch the `KeptnMetric` object to be reconciled.
+   * If the object is not found,
+     it returns and lets Kubernetes handle deleting all associated resources.
+1. If the object is found, the code checks
+   whether the metric has been updated within the interval
+   that is defined in the `spec.fetchintervalseconds` field.
+   * If not, it skips reconciling and requeues the request for later.
+1. If the metric should be reconciled,
+   it fetches the provider defined in the `spec.provider.name` field.
+   * If the provider is not found,
+     it returns and requeues the request for later.
+1. If the provider is found,
+   it loads the provider and evaluates the query
+   defined in the `spec.query` field.
+1. If the evaluation is successful,
+   it stores the fetched value
+   in the metric status field of the `KeptnMetric` object.
+1. If the evaluation fails,
+   the error and reason is also provided in the
+   [MetricStatus](../../../../crd-ref/metrics/v1alpha3/#keptnmetricstatus)
+   resource.
+   The error is described in human-readable language
+   and as raw data to help identify the source of the problem
+   (such as a forbidden code).
+
```

Comment:
I don't think we need to explain step-by-step here how the controller works. If we in the future change something, this will be out-of-sync and if the user wants to know exactly how the controller works he should check the code. I would go here with some general description what the controller does instead of step-by-step description.

Reply:
But isn't the point of an architecture doc to explain exactly how something works?  And yes, this does mean that we may need to update the docs as the product changes.  What does everyone else think?

Reply:
Maybe we can have a slightly more generic description?  

When a new KeptnMetric` is found the controller checks whether the metric has been updated within the interval
  that is defined in the `spec.fetchintervalseconds` field. If not, it skips reconciling and requeues the request for later.
Otherwise it attempts to fetch the provider defined in the `spec.provider.name` field.
It queries the provider with the query defined in the `spec.query` field, than evaluates the result.
If the evaluation is successful, it stores the fetched value in the metric status field of the `KeptnMetric` object.
If the evaluation fails, the error and reason is also provided in the
   [MetricStatus](../../../../crd-ref/metrics/v1alpha3/#keptnmetricstatus)
   resource. The error is described in human-readable language and as raw data to help identify the source of the problem (such as a forbidden code).

Reply:
@odubajDT I think this is generic enough, no? @StackScribe would it be an ok level of detail?

Reply:
done, I think.  Check it out.